test/lib/heap.c

```diff
@@ -177,34 +177,7 @@ void test_heap_setup(const MunitParameter params[], void *user_data)
 /* Ensure we're starting leaving a clean memory behind. */
 void test_heap_tear_down(void *data)
 {
-	sqlite3_mem_methods mem;
```

Comment:
Why are we removing the tear_down completely?

Reply:
The only point of the teardown was to complain if there were outstanding allocations, but that's already commented out because it was buggy/flaky. Without the teardown we leave the faulty allocator in place, but that's okay because it behaves exactly the same as the default allocator unless you ask for a fault explicitly. It's also just impossible to soundly restore the allocator because SQLite bans calling `sqlite3_config(SQLITE_CONFIG_MALLOC, ...)` after any other sqlite3 interface has been used (https://sqlite.org/c3ref/config.html).

Reply:
I sanity-checked myself on this with GDB and I'm wrong---we call `sqlite3_shutdown` in the sqlite fixture, which should make this code okay. Reverted the commit. I think what I was seeing when working on this PR is a pre-existing issue where we don't use the sqlite fixture everywhere we should; I've added a separate commit to fix this in #694 since it was affecting that PR and the fix is quick. Thanks for poking at this!

Reply:
Thanks for looking into it Cole.